### ISSUE TITLE
Help with your module

### ISSUE THREAD

**Issue by soslug** [fewieden/MMM-TTS#1]
Hi I am a noob especially with js code

I understand a little your instructions but the code implementation i seem to be having trouble with.

Each time i add the code snippit following your instructions you do not tell me exactly where to put it.

As a noob I orginally thought oh well everything that whatever goes to the screen goes to tts, but thinking about it a while I decided to incorporate with just one module rather than all modules. I choose the Compliments module to start with

{
    module: 'MMM-TTS',
    position: 'top_right',
    config: {
                 module: 'compliments',
                 position: 'lower_third'
                 }
}

I know I am doing something wrong as the configuration causes the program to ask me to create a config file.

Could you advise some corrective measures I could take to get this to work. I assume it can only work with one of the modules.

**Comment by fewieden** [comment]
I think you misunderstood the functionality of this module. It doesn't read the modules content, a module has to explicitly send a notification with the text to this module.
```
{
  module: "MMM-TTS",
  position: "top_right"
}
```
this is all it needs if you like the default behaviour

**Comment by soslug** [comment]
Ah I certainly did misunderstand, but I am still getting no audio how do I fix that?

**Comment by fewieden** [comment]
you can try:

```
{
  module: "MMM-TTS",
  position: "top_right",
  config: {
    debug: true
  }
}
```

this will show you the messages it should read, so you can check if the audio is the problem or you don't receive the notifications.

**Comment by soslug** [comment]
I am not sure I am looking in the correct place but I found debug in /var/log
this is my output
pi@raspberrypi:/var/log $ tail -f debug 
Feb 18 21:32:06 raspberrypi kernel: [    0.000000]   Normal zone: 2124 pages used for memmap
Feb 18 21:32:06 raspberrypi kernel: [    0.000000]   Normal zone: 0 pages reserved
Feb 18 21:32:06 raspberrypi kernel: [    0.000000]   Normal zone: 241664 pages, LIFO batch:31
Feb 18 21:32:06 raspberrypi kernel: [    0.000000] pcpu-alloc: s22592 r8192 d22464 u53248 alloc=13*4096
Feb 18 21:32:06 raspberrypi kernel: [    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3 
Feb 18 21:32:06 raspberrypi kernel: [    0.789119] dwc_otg: Microframe scheduler enabled
Feb 18 21:32:06 raspberrypi kernel: [    0.880375] dwc_otg: FIQ enabled
Feb 18 21:32:06 raspberrypi kernel: [    0.880385] dwc_otg: NAK holdoff enabled
Feb 18 21:32:06 raspberrypi kernel: [    0.880394] dwc_otg: FIQ split-transaction FSM enabled
Feb 18 21:32:06 raspberrypi kernel: [    0.880430] Module dwc_common_port init

Unfortunately to my untrained eye it offers me no information assuming I am looking in the right place of course.

**Comment by fewieden** [comment]
those are the linux logs, the debug information should be displayed in the mm interface in the position you set in config.

if you want you can contact me on the forum, so we can start a remote desktop session for further help if you like

**Comment by fewieden** [comment]
Does your issue still persist?

**Comment by soslug** [comment]
Hi well, yes I still have no idea how to activate the speech for the magicmirror project although I understand I must implicitly define what I want, I still have no idea how to achieve it. Some examples of the default or common modules used would be very useful. I am still very new to 'js' but would appreciate a nudge or two in the right direction. I am not a complete idiot, but am getting that way !

**Comment by fewieden** [comment]
This is a really simplified module that sends a notification after the mirror is ready to the tts module which should read the message.

config.js
```
{module: 'TTS-Test'}
```

TTS-Test.js
```
Module.register('TTS-Test', {
    notificationReceived(notification) {
        if (notification === 'DOM_OBJECTS_CREATED') {
            this.sendNotification('MMM-TTS', 'This is a text to read. Hello World!');
        }
    }
});
```

**Comment by soslug** [comment]
I feel really stupid asking this and apologise in advance, this looks like this is inserted into the original config.js this were all the other default modules are defined. My question, I have your module installed thats fine I get a logo for want of a better word in the top right hand corner, - Do I use all this text above where do i place it , does it matter, or do you mean new folder new config.js. Sorry what is simple to you is not so simple for me.

**Comment by fewieden** [comment]
the config part goes into the main config file at ~/MagicMirror/config, for the sample module you have to create a new directory, called TTS-Test, in the modules folder ~/MagicMirror/modules. In the new directory you have to create a file called TTS-Test.js with the content from above.

**Comment by soslug** [comment]
I have added the module and rebooted I am of the assumption this is stand alone and would not require an internet connection. that said I have no audio output for this module. Any suggestions is there a log enabled that I can look at. Cheers in advance

**Comment by fewieden** [comment]
Then your problem is somewhere out of scope of this module. I suggest to check the links of the dependencies in the readme file.

**Comment by soslug** [comment]
I could clearly see the text being displayed and confirmed the audio on the pi was working with speaker-test -c2 -t wav I feel I am close to getting it to work. you had an update for MMM-TTS which I have pulled, but still no audio text to speech to speak of. Any advice welcome

**Comment by fewieden** [comment]
you could test manually if your installation of festival is working http://www.festvox.org/docs/manual-2.4.0/festival_7.html#Quick-start

**Comment by soslug** [comment]
Ok that is weird it worked once but cannot repeat definitely loud and clear what next

**Comment by soslug** [comment]
To clarify i used the following
festival
festival> (SayText, "Hello this is test of Festival")
#<Utterance 0x753f99c8>
festival>

This worked only once could not repeat

**Comment by soslug** [comment]
Ok the festival component is indeed working I just needed to extend the sentence past the few words I had written, even so it is only completing to audio the last part of the sentence not the whole of the sentence specified. But I can at least do this repeatedly. So I am guessing the reason I am not getting audio from MMM-TTS is the the sentence to be tested is so very short. I will test this tomorrow if I have time to spare.

**Comment by soslug** [comment]
For the first time I can confirm that module is working although I have indeed needed to lengthen the sentence and then only received the tail of the sentence as audio. Last two words actually!

**Comment by fewieden** [comment]
Ok so i ran your test on my rpi3
```
festival
festival> (SayText, "Hello this is test of Festival")
```
which I get the response ```SIOD ERROR: comma-not-inside-backquote```
so I ran
```festival> (SayText "Hello this is test of Festival")```
which works fine and can be repeated as much as I like. I can also hear the full sentence and not only the last parts.

You probably should try to reinstall festival

**Comment by soslug** [comment]
I have removed festival for rpi3 and reinstalled unfortunately i get the same problem. The only thing that might be different about yours and mine is that I am using the latest raspbian os lite rather than the full packed version, another thing that might be a factor is that i am using pixel with lightdm. But as full blown raspbian uses pixel and lightdm I doubt that has a bearing.

**Comment by soslug** [comment]
With you example code above I used exactly the same words I get no error but also no sound at all only a click. However I extended the line by one word Hello and that was all I got hello not the whole sentence. Repeating the same I am back to a click again. extending by one word this time i got "festival hello this" as speech attempting to repeat I am back to a click on speaker.

**Comment by fewieden** [comment]
I have running pixel as well. but this issue is going to be more related to festival than my actual module. I suggest to search for a github repo or forum of festival where are some experts sitting and ask them, they can help you more I guess.

**Comment by soslug** [comment]
ok thanks

**Comment by fewieden** [comment]
I will close for now, but you can respond here if you find a solution.
